Lintian says, these do no belong the FreeDesktop specs
--- a/k9copy.desktop
+++ b/k9copy.desktop
@@ -1,6 +1,4 @@
-#!/usr/bin/env xdg-open
 [Desktop Entry]
-Encoding=UTF-8
 Type=Application
 Name=k9copy
 Exec=k9copy -caption "%c" %i %m  %u
@@ -14,7 +12,6 @@
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
 GenericName[uk]=Копіювання відео-DVD
-MimeTypes=
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
--- a/k9copy_assistant.desktop
+++ b/k9copy_assistant.desktop
@@ -1,6 +1,4 @@
-#!/usr/bin/env xdg-open
 [Desktop Entry]
-Encoding=UTF-8
 Type=Application
 Name=k9copy assistant
 Exec=k9copy --assistant -caption "%c" %i %m  %u
@@ -14,7 +12,6 @@
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
 GenericName[uk]=Копіювання відео-DVD
-MimeTypes=
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
